### DISCUSSION TITLE
Adding common tags to all metrics

### DISCUSSION THREAD

**Discussion by stefano-salmaso** [open-telemetry/opentelemetry-java-instrumentation#8719] [Q&A] [answered]
Hi all!
We are using opentelemetry java-agent and we need to add some common tags to all the metrics generated by our application.
The value is fixed and defined at startup.
We are able to add common tags to the Micrometer MeterRegistry.. but we need the same tags also to the metric generated directly by the auto instrumentation, like Tomcat metrics.
We tried in several way but with no luck!
Adding metrics using the collector is not an option for us.

Any way to do that?
Thanks

**Comment by laurit** [comment]
You could try using a resource attribute https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#opentelemetry-resource

**Comment by stefano-salmaso** [comment]
Hi @laurit 
I tried... but the resources added are not present as tags into the metrics!

**Reply by mateuszrzeszutek** [reply]
Hey @stefano-salmaso ,

Are you perhaps using the prometheus metrics exporter?

**Reply by stefano-salmaso** [reply]
I tried both with prometheus and otlp (to the collector) but again... nothing I added with `otel.resource.attributes` is present into metrics

**Reply by mateuszrzeszutek** [reply, marked as answer]
OTLP definitely captures all resource attributes -- if you're not seeing them in your chosen observability backend it may be that it does not support resource attributes.

The Prometheus exporter unfortunately has a limited support for resource attributes, see https://github.com/open-telemetry/opentelemetry-java/issues/5529

**Reply by stefano-salmaso** [reply]
Hi @mateuszrzeszutek 
with OTLP exporter I have all resource attributes as tags into metrics. My point was the `resource_to_telemetry_conversion` with false in the collector. Changing it everything is working!
Thanks!
